build.gradle

```diff
@@ -840,6 +858,16 @@ tasks.register("downloadJdk", Download) {
     }
 }
 
+tasks.register("checkNewJdkVersion") {
+    def versionYml = new Yaml().load(new File("$projectDir/versions.yml").text)
+
+    // use Linux x86_64 as canary platform
+    def jdkDetails = new JDKDetails(versionYml, "linux", "x86_64")
+    if (!jdkDetails.isLocalVersionMatchingLatest()) {
+        throw new GradleException("New version available for major: ${jdkDetails.major}")
```

Comment:
nitpick
Can we throw the exception with instruction? 
"Found new jdk version. Please update `version.yml` to ${newJdkDetails.revision} build ${newJdkDetails.build}"

Reply:
Updated the isLocalVersionMatchingLatest method to become a check method and throw directly the exception, in this way it can provide the new version and build number to use.